pkg/asset/installconfig/gcp/validation.go

```diff
@@ -469,31 +470,37 @@ func validateRegion(client API, ic *types.InstallConfig, fieldPath *field.Path)
 // ValidateCredentialMode checks whether the credential mode is
 // compatible with the authentication mode.
 func ValidateCredentialMode(client API, ic *types.InstallConfig) error {
-	creds := client.GetCredentials()
-
-	if creds.JSON == nil && ic.CredentialsMode != types.ManualCredentialsMode {
-		errMsg := "environmental authentication is only supported with Manual credentials mode"
-		return field.Forbidden(field.NewPath("credentialsMode"), errMsg)
-	}
-
-	return nil
+	allErrs := validateCredentialMode(client, ic)
```

Comment:
I don't see why have 2 methods ValidateCredentialMode() and validateCredentialMode()? Just for the different